waiter/integration/waiter/autoscaling_test.clj

```diff
@@ -15,55 +15,41 @@
             [waiter.client-tools :refer :all]
             [waiter.utils :as utils]))
 
-(defn- scaling-for-service-test [testing-str waiter-url threads requests-per-thread request-fn]
+(defn- scaling-for-service-test [testing-str waiter-url target-instances concurrency-level request-fn]
   (testing testing-str
-    (let [expect-tasks-running (fn [service-id pred msg]
-                                 (loop [i 1
-                                        tasks-running 0]
-                                   (if (< i (* 5 4 requests-per-thread))
-                                     (let [tasks-running (num-tasks-running waiter-url service-id :prev-tasks-running tasks-running)]
-                                       (when (not (pred tasks-running))
-                                         (Thread/sleep 1000)
-                                         (recur (inc i) (int tasks-running))))
-                                     (is nil msg))))]
-      (let [_ (log/info (str "Making canary request..."))
-            first-request (request-fn)
-            service-id (retrieve-service-id waiter-url (:request-headers first-request))]
-        (expect-tasks-running service-id #(= % 1) "Never saw initial 1 task running")
-        (future (dorun (pmap (fn [_] (dotimes [_ requests-per-thread] (request-fn))) (range threads))))
-        (expect-tasks-running service-id #(> % (/ threads 2)) (str "Never scaled up to " (/ threads 2) " task(s)"))
-        (expect-tasks-running service-id #(= % 1) "Never scaled down to 1 task")
-        (delete-service waiter-url service-id)))))
+    (let [continue-running (atom true)
+          first-request (request-fn)
+          service-id (retrieve-service-id waiter-url (:request-headers first-request))
+          count-instances (fn [] (num-instances waiter-url service-id))]
```

Comment:
`num-instances` can still suffer from flakes if marathon gets busy. An alternate approach might be to take look at `[:autoscaler-state :target-instances]` from `/state/<service-id>` endpoint.

Reply:
If this had been the test code, then the test would have passed with Marathon even though nothing was being scaled down.

Reply:
I think he issue is that Marathon might not scale up to `target-instances` if it is low on free resources.

Reply:
That's one of the reasons I changed it to 3 instead of 5.  I'd like to avoid another future regression. I'm inclined to keep this a "full" integration test, including making sure Marathon scales all the way up and all the way down, with real tasks.